packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -152,12 +160,12 @@ function fromSdkOperationParameters(
   const parameters = new Map<SdkHttpParameter, InputParameter>();
   for (const p of operation.parameters) {
     if (p.kind === "cookie") {
-      Logger.getInstance().error(
-        `Cookie parameter is not supported: ${p.name}, found in operation ${operation.path}`,
-      );
-      throw new Error(
-        `Cookie parameter is not supported: ${p.name}, found in operation ${operation.path}`,
-      );
+      reportDiagnostic(sdkContext.program, {
```

Comment:
could we add a method to our logger to make it be able to report diagnostic so that we do not have to include the `sdkContext.program` parameter?

Reply:
Yes, I will file an issue to rework the logger API so that it can be used to wrap reportDiagnostics.

Reply:
Filed https://github.com/microsoft/typespec/issues/5921